pallets/runtime/src/bridge.rs

```diff
@@ -391,4 +405,73 @@ impl<T: Trait> Module<T> {
         }
         Ok(None)
     }
+
+    /// Handles a bridge transaction proposal immediately.
+    fn handle_bridge_tx_now(bridge_tx: BridgeTx<T::AccountId, T::Balance>) -> DispatchResult {
+        if let Some(PendingTx { did, bridge_tx }) = Self::issue(bridge_tx.clone())? {
+            <PendingTxs<T>>::mutate(did, |txs| txs.push(bridge_tx.clone()));
+            Self::deposit_event(RawEvent::Pending(PendingTx { did, bridge_tx }));
+        } else {
+            <HandledTxs<T>>::insert(&bridge_tx, true);
+            Self::deposit_event(RawEvent::Bridged(bridge_tx));
+        }
+        Ok(())
+    }
+
+    /// Handles a bridge transaction proposal after `timelock` blocks.
+    fn handle_bridge_tx_later(
+        bridge_tx: BridgeTx<T::AccountId, T::Balance>,
+        timelock: T::BlockNumber,
+    ) -> DispatchResult {
+        let current_block_number = <system::Module<T>>::block_number();
+        let mut unlock_block_number = current_block_number + timelock;
+        let range = <T::BlockNumber>::from(BLOCK_RANGE_FOR_TIMELOCK);
+        let max_unlock_block_number = unlock_block_number + range - One::one();
+        while Self::timelocked_txs(unlock_block_number).len() >= MAX_TIMELOCKED_TXS_PER_BLOCK
+            && unlock_block_number <= max_unlock_block_number
+        {
+            unlock_block_number += One::one();
+        }
+        ensure!(
+            unlock_block_number <= max_unlock_block_number,
+            Error::<T>::TimelockBlockRangeFull
+        );
+        <TimelockedTxs<T>>::mutate(unlock_block_number, |txs| {
+            txs.push(bridge_tx);
+        });
+        Ok(())
+    }
+
+    /// Handles the timelocked transactions that are set to unlock at the given block number.
+    fn handle_timelocked_txs(block_number: T::BlockNumber) {
+        let txs = <TimelockedTxs<T>>::take(block_number);
+        for tx in txs {
+            if let Err(e) = Self::handle_bridge_tx_now(tx) {
+                sp_runtime::print(e);
```

Comment:
Under what circumstances can handle_bridge_tx_now return an error?

Reply:
When `issue` returns an error:
* Incorrect account address given as the issue recipient (`AccountKey::try_from` error `"Invalid size for a key"`).
* The target account or identity balance overflows if `amount` is added (same error for both `resolve_into_existing` and `resolve_into_existing_identity`).